src/components/Table/TableToolbar/TableToolbar.jsx

```diff
@@ -143,115 +143,132 @@ const TableToolbar = ({
-        ) : null}
+}) => {
+  const [currentSearchValue, setCurrentSearchValue] = useState(search?.defaultValue || '');
 
-        {// Default card header actions should be to the right of the table-specific actions
-        customToolbarContent || null}
-      </TableToolbarContent>
-    )}
-  </CarbonTableToolbar>
-);
+  return (
+    <CarbonTableToolbar className={classnames(`${iotPrefix}--table-toolbar`, className)}>
+      <TableBatchActions
+        className={`${iotPrefix}--table-batch-actions`}
+        onCancel={onCancelBatchAction}
+        shouldShowBatchActions={hasRowSelection === 'multi' && totalSelected > 0}
+        totalSelected={totalSelected}
+        translateWithId={(...args) => tableTranslateWithId(i18n, ...args)}
+      >
+        {batchActions.map(({ id, labelText, ...others }) => (
+          <TableBatchAction key={id} onClick={() => onApplyBatchAction(id)} {...others}>
+            {labelText}
+          </TableBatchAction>
+        ))}
+      </TableBatchActions>
+      {secondaryTitle ? (
+        // eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for
+        <label className={`${iotPrefix}--table-toolbar-secondary-title`}>{secondaryTitle}</label>
+      ) : null}
+      {// Deprecated in favor of secondaryTitle for a more general use-case
+      hasRowCountInHeader ? (
+        // eslint-disable-next-line jsx-a11y/label-has-associated-control, jsx-a11y/label-has-for
+        <label className={`${iotPrefix}--table-toolbar-secondary-title`}>
+          {i18n.rowCountInHeader(totalItemsCount)}
+        </label>
+      ) : null}
+      {tooltip && (
+        <div className={`${iotPrefix}--table-tooltip-container`}>
+          <Tooltip
+            triggerId={`card-tooltip-trigger-${tableId}`}
+            tooltipId={`card-tooltip-${tableId}`}
+            triggerText=""
+          >
+            {tooltip}
+          </Tooltip>
+        </div>
+      )}
+      {activeBar === 'rowEdit' ? (
+        <div className={`${iotPrefix}--table-row-edit-actions`}>{rowEditBarButtons}</div>
+      ) : (
+        <TableToolbarContent className={`${iotPrefix}--table-toolbar-content`}>
+          {hasSearch ? (
+            <TableToolbarSearch
+              {...search}
+              // defaultValue is updated when onApplySearch is called. If the defaultValue does
+              // not match the currentSearchValue, we can infer that a new defaultValue has been
+              // set externally and we should render a new instance of the component.
+              // https://github.com/IBM/carbon-addons-iot-react/issues/1386
+              key={
+                ((search.defaultValue || search.defaultValue === '') &&
+                  search.defaultValue !== currentSearchValue) ||
+                ((search.value || search.value === '') && search.value !== currentSearchValue)
+                  ? `${search.defaultValue}${search.value}`
+                  : `${tableId}-toolbar-search`
+              }
```

Comment:
Unfortunately the diff for this component is awful because I had to remove the implicit return to use useState, but this is what actually fixes the problem.

Reply:
thanks for the comment this will help us understand why in the future, I believe the !isEmpty(search.value) from lodash will check the search.value || search.value === '' in a easy to read way